src/Moq/InvocationShape.cs

```diff
@@ -109,5 +110,71 @@ private bool IsOverride(MethodInfo invocationMethod)
 
 			return true;
 		}
+
+		public bool Equals(InvocationShape other)
+		{
+			if (this.Method != other.Method)
+			{
+				return false;
+			}
+
+			if (this.Arguments.Count != other.Arguments.Count)
```

Comment:
This could probably be just an assertion. The argument count should always be equal if the methods are already the same.